### PR TITLE
Add basic component wrappers

### LINKED ISSUE #25: [Feature request] Add wrappers for Message Components
**Is your feature request related to a problem? Please describe.**
While using the Message Components API exposed by `nyxx_interactions` is very flexible, it is often verbose and does not integrate well with `nyxx_commands`.

**Describe the solution you'd like**
Add wrappers for common use cases of Message Components that integrate with `nyxx_commands`.

### PR DESCRIPTION
**Pull request by abitofevrything**
# Description

Added basic wrappers for components. More might be added in the future if enough demand is found.

Closes #25.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Ran `dart analyze .`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/src/context/component_wrappers.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          .map((event) => event..acknowledge())
+          .where(
+            (event) =>
+                !authorOnly ||
+                (event.interaction.memberAuthor ?? event.interaction.userAuthor as SnowflakeEntity)
+                        .id ==
+                    user.id,
+          )
+          .timeout(
+            timeout ?? Duration(),
```
Comment (l7ssha) [inline lib/src/context/component_wrappers.dart:41]: what would happen if timeout is null? timeout is gonna be instant?

Comment (l7ssha) [reply lib/src/context/component_wrappers.dart:41]: kinda weird that timeout has default value but its nullable anyway, whats the use case?

File: lib/src/context/component_wrappers.dart [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    List<ButtonBuilder> buttons = [
+      ButtonBuilder(confirmMessage, randomAlpha(10), ComponentStyle.success),
+      ButtonBuilder(denyMessage, randomAlpha(10), ComponentStyle.danger),
+    ];
+
+    componentMessageBuilder.addComponentRow(ComponentRowBuilder()
+      ..addComponent(buttons[0])
+      ..addComponent(buttons[1]));
```
Comment (l7ssha) [inline lib/src/context/component_wrappers.dart:76]: thats kinda weird constructed. whats the point of this array if you everywhere refer to them by index? at this points its better to create two local variables

File: lib/src/context/component_wrappers.dart [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    await respond(componentMessageBuilder);
+
+    IButtonInteractionEvent event = await getButtonPress(buttons);
+
```
Comment (l7ssha) [inline lib/src/context/component_wrappers.dart:81]:
```suggestion
```

**Review by l7ssha — CHANGES_REQUESTED** [review]

File: lib/src/context/component_wrappers.dart (hunk 1 above)
Comment (abitofevrything) [reply lib/src/context/component_wrappers.dart:41]: A null timeout will indicate that the user does not want the interaction awaiting to timeout.

In that case, a default timeout of 0 is passed to the `timeout` method on the stream. However, what matters is the `onTimeout` function provided: if the user-provided timeout is `null`, then an empty function is passed which causes the stream to continue as normal.

However, in most cases users will want to respond using the current interaction, so a timeout is necessary that avoid token expiration. If the user wants no timeout, they have to be explicit and manually set `timeout` to null.

**Review by l7ssha — APPROVED** [review]

**Pull request merged**